### ISSUE TITLE
Dashboards for aggregated results (perf)

### ISSUE THREAD

**Issue by elnyry** [mojaloop/project#1000]
## **Goal**:
As _Hub Tech Ops_
I want to be able to _view aggregations of various numbers / metriocs of transactions_
so that _I can use them for managing a Mojaloop Hub and for verifying its performance_

**Tasks**:
- [x] Create a custom dashboard to view aggregated numbers (PoC)
- [x] Discuss with team, review
- [x] Incorporate feedback and release
- [ ] Validate with sample runs

**Notes**:
- Troubleshoot shard index issue on apm-* index -- New story
- Create story to investigate filter by labels (i.e. transactionId) issue -- New story
- Figure out why we are missing other trace flows (e.g. prepare) -- Hold-off till release / validate
- Look into eventstreamprocessor <-- handle throttling from EFK -- New story
- (For future development) Look at number of error and error breakdown tiles to trigger alerts for specific error codes -- New story

**Acceptance Criteria**:
- [ ] A custom dashboard (or multiple) that helps view aggregated metrics, numbers on the Mojaloop Switch is available, that can be used for Hub TechOps and/or Performance testing as needed

## **Pull Requests**:
- [ ] TBD

## **Follow-up**:
- N/A

**Dependencies**:
- N/A

## **Accountability**:
- Owner: TBC
- QA/Review: TBC

## **Notes**:
The Mojaloop implementation of the distributed tracing logs everything under the same service (currently is called event-stream-processor since the EFK agent is taking the default name from the package.json)

Our understanding is that single transfer request (end-to-end) should be represented as single trace with multiple spans. So our spans would be:

1. ml-api-adapter receives POST request to create new transfer. The transfer is validated and then transfer prepare message is produced to related kafka topic. This whole process is represented as a single span - ml_transfer_prepare. 
2. Next the central-ledger prepare handler picks up the message, validates it etc and produces new message for position adjustments to the kafka topic. Again the whole process is another span - cl_transfer_prepare. That span is child span of the previous and has the same trace.id
3. The central-ledger position handler work is another span - cl_transfer_position, which again is a child span of the previous one and has the same trace.id
4. Until the flow reaches the ml-api-adapter notification handler, the spans are children of the previous and basically we have a waterfall process.
5. If error happens somewhere in the process it is visualized nicely and can be seen in the dashboard, as well. 

In the APM dashboard each span represents single transaction and since the dashboard is transaction-centric, we have very good metrics for transactions duration, transactions per minute and other transactions statistics. What we lack, though, are metrics for end-to-end transfers, which is vital for us.

The total time is visualized on the default APM dashboard, only on a bar above the waterfall representation of the spans. You cannot sort, search or do anything with that numbers. 

To tackle this issue, we can create either a custom dashboard, that shows end-to-end transfer by summing the transaction.duration.us fields for all transactions with either same labels.transactionId field or trace.id field or we can programmatically create a master span, that covers the transfer end to end and work with it in the default APM dashboard.

Looks like several paid solutions touching the same issue are available, but nothing in the open source universe. Also some articles suggested using APM for bird overview and using another dashboard as Zipkin or Jaeger for the details drill down.

**Comment by vgenev** [comment]
Will continue after #1013

**Comment by vgenev** [comment]
![image](https://user-images.githubusercontent.com/5424739/68483148-b0c53500-0243-11ea-9acf-a7a5bc50c3ab.png)
